core/vault.go

```diff
@@ -0,0 +1,416 @@
+		return err
+	}
+
+	encType, err := vault.encrypt(key[:], []byte(creds.EncryptionType))
+	if err != nil {
+		log.Errorf("Failed to encrypt type for longterm storage: %s", err)
+		return err
+	}
+
+	encryptedCreds := VaultCreds{
+		SealKey:        encSeal,
+		RootToken:      encRoot,
+		EncryptionType: encType,
+	}
+
+	log.Debugf("marshaling credentials for longterm storage")
+	b, err := json.Marshal(encryptedCreds)
+	if err != nil {
+		log.Errorf("failed to marshal vault root token / seal key for longterm storage: %s", err)
+		return err
+	}
+	log.Debugf("storing credentials at %s (mode 0600)", store)
+	err = ioutil.WriteFile(store, b, 0600)
+	if err != nil {
+		log.Errorf("failed to write credentials to longterm storage file %s: %s", store, err)
+		return err
+	}
+	return nil
+}
+
+func (vault *Vault) Gen(length int) (string, error) {
+	chars := "0123456789ABCDEF"
+	var buffer bytes.Buffer
+
+	for i := 0; i < length; i++ {
+		if i > 0 && i%4 == 0 {
+			buffer.WriteString("-")
+		}
+		index, err := rand.Int(rand.Reader, big.NewInt(int64(len(chars))))
+		if err != nil {
+			return "", err
+		}
+		indexInt := index.Int64()
+		buffer.WriteString(string(chars[indexInt]))
+	}
+
+	return buffer.String(), nil
+}
+
+func (vault *Vault) IsSealed() (bool, error) {
+	res, err := vault.Do("GET", "/v1/sys/seal-status", nil)
+	if err != nil {
+		log.Errorf("failed to check current seal status of the vault: %s", err)
+		return true, err
+	}
+	b, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		log.Errorf("failed to read response from the vault, concerning current seal status: %s", err)
+		return true, err
+	}
+
+	err = json.Unmarshal(b, &status)
+	if err != nil {
+		log.Errorf("failed to parse response from the vault, concerning current seal status: %s", err)
+		return true, err
+	}
+
+	return status.Sealed, err
+}
+
+func (vault *Vault) IsInitialized() (bool, error) {
+	res, err := vault.Do("GET", "/v1/sys/init", nil)
+	if err != nil {
+		log.Errorf("failed to check initialization state of the vault: %s", err)
+		return false, err
+	}
+	b, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		log.Errorf("failed to read response from the vault, concerning its initialization state: %s", err)
+		return false, err
+	}
+	var init struct {
+		Initialized bool `json:"initialized"`
+	}
+	if err = json.Unmarshal(b, &init); err != nil {
+		log.Errorf("failed to parse response from the vault, concerning its initialization state: %s", err)
+		return false, err
+	}
+	return init.Initialized, err
+}
+
+func (vault *Vault) encrypt(key, text []byte) (string, error) {
+	block, err := aes.NewCipher(key)
+	if err != nil {
+		return "", err
+	}
+	ciphertext := make([]byte, aes.BlockSize+len(text))
+	iv := ciphertext[:aes.BlockSize]
+	if _, err := io.ReadFull(rand.Reader, iv); err != nil {
+		return "", err
+	}
+	cfb := cipher.NewCFBEncrypter(block, iv)
+	cfb.XORKeyStream(ciphertext[aes.BlockSize:], text)
+	return base64.StdEncoding.EncodeToString(ciphertext), nil
+}
```

Comment:
If we are always using AES-256/CFB for the credentials file, we _may_ want to document that somewhere.